src/templates/hl7v2/ADT_A01.hbs

```diff
@@ -3,50 +3,69 @@
     "type": "transaction",
     "entry": [
         {{#with (getFirstSegments msg.v2 'PID' 'PD1' 'PV1' 'PV2' 'PR1' 'AVR')}}
-    
-            {{>Resources/Patient.hbs PID=PID PD1=PD1 NK1=NK1 ID=(generateUUID PID-3-1)}},
-            {{>Resources/Encounter.hbs PV1=PV1 PV2=PV2 ID=(generateUUID PV1)}},
-            {{>Resources/Procedure.hbs PR1=PR1 ID=(generateUUID PR1)}},
-            {{>References/Encounter/subject.hbs ID=(generateUUID PV1) REF=(concat 'Patient/' (generateUUID PID-3-1))}}, 
-            {{>References/Procedure/subject.hbs ID=(generateUUID PR1) REF=(concat 'Patient/' (generateUUID PID-3-1))}},	
-            
+
+            {{#if PID-3-1}},
+                {{>Resources/Patient.hbs PID=PID PD1=PD1 NK1=NK1 ID=(generateUUID PID-3-1)}},
+            {{/if}},
+
+            {{#if PV1}},
+                {{>Resources/Encounter.hbs PV1=PV1 PV2=PV2 ID=(generateUUID PV1)}},
+                {{>References/Encounter/subject.hbs ID=(generateUUID PV1) REF=(concat 'Patient/' (generateUUID PID-3-1))}},
```

Comment:
Are we checking the existence of REF (PID-3-1) here?

Reply:
We only concentrate on the "ID" property, because the incorrection of "ID" means the whole resource is useless. But the incorrection of "REF" just means there is a single blank property and will not has such influences, just ignore it currently.

Reply:
Update: According to the suggestions from Ranvijay, the "REF" attribtues are also very important, so I will fixed it too.